diff --git a/matlab.el b/matlab.el
index b289efc..2f18273 100644
--- a/matlab.el
+++ b/matlab.el
@@ -434,6 +434,14 @@ Valid values are:
 
 (make-variable-buffer-local 'matlab-return-add-semicolon)
 
+(defcustom matlab-change-current-directory nil
+  "*If non nil, make file's directory the current directory when
+evaluating it."
+  :group 'matlab
+  :type 'boolean)
+
+(make-variable-buffer-local 'matlab-change-current-directory)
+
 ;; Load in the region we use for highlighting stuff.
 (if (and (featurep 'custom) (fboundp 'custom-declare-variable))
 
@@ -4969,6 +4977,8 @@ Similar to  `comint-send-input'."
   (let ((fn-name (file-name-sans-extension
 		  (file-name-nondirectory (buffer-file-name))))
 	(msbn (concat "*" matlab-shell-buffer-name "*"))
+        (dir (file-name-directory buffer-file-name))
+        (change-cd matlab-change-current-directory)
 	(param ""))
     (save-buffer)
     ;; Do we need parameters?
@@ -4991,6 +5001,21 @@ Similar to  `comint-send-input'."
 	  (select-window (get-buffer-window msbn t))
 	(switch-to-buffer (concat "*" matlab-shell-buffer-name "*")))
 
+      ;; change current directory?
+      (if change-cd
+          (let ((cmd (progn
+                       (mapc
+                        (lambda (e)
+                          (while (string-match (car e) dir)
+                            (setq dir (replace-match
+                                       (format "', char(%s), '" (cdr e)) t t dir))))
+                        '(("ô" . "244")
+                          ("é" . "233")
+                          ("è" . "232")
+                          ("à" . "224")))
+                       dir)))
+            (matlab-shell-send-string (concat "cd(['" cmd "'])\n"))))
+
       (let ((cmd (concat fn-name " " param)))
 	(matlab-shell-add-to-input-history cmd)
 	(matlab-shell-send-string (concat cmd "\n"))))))
