diff -u /home/karthik/.local/share/git/matlab-emacs-src/matlab-shell.el /home/karthik/.local/share/git/matlab-emacs-src/matlab-shell-new.el
--- ./matlab-emacs-src/matlab-shell.el	2021-07-18 22:49:53.525471052 -0700
+++ ./matlab-emacs-src/matlab-shell-new.el	2021-07-25 22:41:19.801839800 -0700
@@ -2413,13 +2413,12 @@
       (goto-char (point-min))
       (dolist (F functions)
 	(save-excursion
-	  (when (re-search-forward (semantic-tag-name F) nil t)
-	    ;; Found, copy it in.
-	    (let ((ft (matlab-semantic-tag-text F orig)))
-	      (goto-char (point-max))
-	      (insert "% Copy of " (semantic-tag-name F) "\n\n")
-	      (insert ft)
-	      (insert "\n%%\n"))))
+	  ;; Copy all local functions to script.
+	  (let ((ft (matlab-semantic-tag-text F orig)))
+	    (goto-char (point-max))
+	    (insert "% Copy of " (semantic-tag-name F) "\n\n")
+	    (insert ft)
+	    (insert "\n%%\n")))
 	)
 
       ;; Save buffer, and setup ability to run this new script.

Diff finished.  Sun Jul 25 22:41:51 2021
