diff --git a/elisp/matlab-emacs/matlab.el b/elisp/matlab-emacs/matlab.el
index 63ce838..71b6ff0 100755
--- a/elisp/matlab-emacs/matlab.el
+++ b/elisp/matlab-emacs/matlab.el
@@ -3731,6 +3731,6 @@ If ARG is nil, then highlighting is toggled."
 			  post-command-hook)
 		  -1 1)))
     (if (> arg 0)
-	(add-hook 'post-command-hook 'matlab-start-block-highlight-timer)
-      (remove-hook 'post-command-hook 'matlab-start-block-highlight-timer))))
+	(add-hook 'post-command-hook 'matlab-start-block-highlight-timer nil :local)
+      (remove-hook 'post-command-hook 'matlab-start-block-highlight-timer :local))))
 
